hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/avro/HoodieAvroParquetInputFormat.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.hadoop.avro;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.ArrayWritable;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.parquet.hadoop.ParquetInputFormat;
+import org.apache.parquet.hadoop.util.ContextUtil;
+
+import java.io.IOException;
+
+public class HoodieAvroParquetInputFormat extends ParquetInputFormat<ArrayWritable> {
```

Comment:
How will this work with Presto/Trino? Do we need to use the annotation `UseRecordReaderFromInputFormat`?

Reply:
@codope 
The question you have considered is very good. I might have been thoughtless.
This PR mainly focus on hive. It also tests and verifies that spark reads hive tables. I don't know much about presto/trino. I hope @XuQianJin-Stars can help verify this and give feedback on the follow-up issues/prs of more aspects of presto/trino. Thanks ~

Reply:
@codope 
Hi~ I've created a new issue in jira https://issues.apache.org/jira/browse/HUDI-4274
After this PR is resolved, I will complete the test/fix in Presto/Trino before version 0.12